### ISSUE TITLE
问题反馈

### ISSUE THREAD

**Issue by mathsmatics** [mumuy/pacman#7]
老哥，游戏界面什么的都做得可以，就是一点，按键有点难按，非得卡着那个点按下按键才能拐弯，可以优化一下这一点

**Comment by mumuy** [comment]
不是一个点，是1/2单元格的空间

**Comment by mathsmatics** [comment]
速度有点快，稍微按早了就没响应，可以放宽点，提前就确定要转的方向

**Comment by rogerclarkmelbourne** [comment]
Normally, you can define the next direction before the next turning point.

e.g. When moving upwards, if you press left, then at the next time that a left turn is possible, then the player will turn left.

Google translate says...

通常，您可以在下一个转折点之前定义下一个方向。

例如 向上移动时，如果你按下左键，那么在下一次可以左转的时候，玩家就会左转。

**Comment by mumuy** [comment]
Actually, I did. I define the next direction before the next at 1/2 cell position.

**Comment by rogerclarkmelbourne** [comment]
OK. 

The existing version did not seem to accept a new direction until at the junction.

**Comment by mumuy** [comment]
'this.control.orientation' is equivalent to the property 'this.requestedOrientation' you submitted.

**Comment by rogerclarkmelbourne** [comment]
something seems to constantly clear   

this.control.orientation

**Comment by rogerclarkmelbourne** [comment]
Sorry.

I think I was confused

Its player.control.orientation which goes back to being undefined.

But I think with this.control.orientation, you code immediatly checks if the direction is invalid, and if it is unvalid it is set to the current direction, so the input from player.control.orientation is lost :-(

**Comment by rogerclarkmelbourne** [comment]
I think this area of code changes the orientation

https://github.com/mumuy/pacman/blob/master/index.js#L869-L877

**Comment by mumuy** [comment]
Yes, The 'update' function updates the coordinates of an object.  'this.control.orientation'  is used to store control parameters, meaning to cache the task that the object will execute
